mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -164,9 +164,48 @@ impl TendermintConf {
     }
 }
 
+struct TendermintRpcClient(AsyncMutex<TendermintRpcClientImpl>);
+
+struct TendermintRpcClientImpl {
+    rpc_clients: Vec<HttpClient>,
+}
+
+#[async_trait]
+impl RpcCommonOps for TendermintCoin {
+    type RpcClient = HttpClient;
+    type Error = TendermintCoinRpcError;
+
+    async fn get_live_client(&self) -> Result<Self::RpcClient, Self::Error> {
+        let mut client_impl = self.client.0.lock().await;
+        let mut new_clients = client_impl.rpc_clients.clone();
+        for (i, client) in client_impl.rpc_clients.iter().enumerate() {
+            if client
+                .perform(HealthRequest)
+                .timeout(Duration::from_secs(3))
+                .await
+                .is_ok()
+            {
+                let res_client = client.clone();
+                // if the first client is alive, no need to change rpc_clients field
+                return if i == 0 {
+                    Ok(res_client)
+                } else {
+                    let len = new_clients.len();
+                    // move the last len - i elements to front, so i client will be the first
+                    new_clients.rotate_right(len - i);
+                    client_impl.rpc_clients = new_clients;
+                    Ok(res_client)
+                };
+            }
+        }
+        return Err(TendermintCoinRpcError::RpcClientError(
+            "All the current rpc nodes are unavailable.".to_string(),
+        ));
+    }
```

Comment:
> get_live_client will need some refactoring too by using rotate_right to rotate clients to make the found live client the first one in the vec / current.

I made a small mistake, I think `rotate_left` is more fitting here to send the non working clients to the end of the `vec`. Then, this function can be more simplified as follows
```rust
async fn get_live_client(&self) -> Result<Self::RpcClient, Self::Error> {
    let mut client_impl = self.client.0.lock().await;
    // try to find first live client
    for (i, client) in client_impl.rpc_clients.clone().iter().enumerate() {
        if client.perform(HealthRequest).await.timeout(Duration::from_secs(3)).is_ok() {
            // Bring the live client to the front of rpc_clients
            client_impl.rpc_clients.rotate_left(i);
            return Ok(client.clone());
        }
    }
    return Err(TendermintCoinRpcError::RpcClientError(
        "All the current rpc nodes are unavailable.".to_string(),
    ));
}
```

Reply:
thanks! done